relay-general/src/pii/attachments.rs

```diff
@@ -314,7 +310,7 @@ impl<'a> Iterator for WStrSegmentIter<'a> {
                 WStr::from_utf16le_unchecked_mut(std::slice::from_raw_parts_mut(ptr, len))
             };
 
-            if encoded.chars().count() < MIN_STRING_LEN {
+            if encoded.chars().take(MIN_STRING_LEN).count() < MIN_STRING_LEN {
```

Comment:
Note that this is slower than just doing `.count()` since you have an optimized count implementation that does not decode.

Reply:
ugh.  can we have a `.count_up_to(5)` iterator method? ;)

but this is a hard call, in general my guess is that fully decoding 5 characters is faster that partially decoding 100+ chars and that we're more likely to have 100+ chars here.

Reply:
You can keep this implementation. In most cases where long strings are detected, doing `.take().count()` will be faster due to the early exit. I am a bit curious if the compiler can see through the inlined `.count()`, but it's probably not worth going that deep. 

Sorry for the distraction :)